src/value/mod.rs

```diff
@@ -858,6 +858,30 @@ impl Value {
     pub fn take(&mut self) -> Value {
         mem::replace(self, Value::Null)
     }
+
+    /// Recursively sorts the contents of the `Value`
+    ///
+    /// ```
+    /// let mut v = json!({ "y": null ,"x": [2.0, 1.0] })
+    /// v.sort()
+    /// assert_eq!(v, json!({ "x": [1.0, 2.0], "y": null }))
+    /// ```
+    pub fn sort(&mut self) -> Value {
+        match self {
+            Value::Null => Value::Null,
+            Value::Bool(a) => Value::Bool(*a),
+            Value::Number(a) => Value::Number(*a),
+            Value::String(a) => Value::String(*a),
```

Comment:
This works if I change these to `.clone()` instead of dereferencing them with `*`, but I don't understand why that's necessary.

```suggestion
            Value::Number(a) => Value::Number(a.clone()),
            Value::String(a) => Value::String(a.clone()),
```

Reply:
It looks like you probably want to be mutating `self` in-place, rather than returning a new `Value`:
```rust
pub fn sort(&mut self) {
    match self {
        Value::Array(ref mut vec) => {
            vec.iter_mut().for_each(Value::sort);
            vec.sort();
        }
        Value::Object(ref mut map) => {
            map.values_mut().for_each(Value::sort);
        }
        _ => {}
    }
}
``